metadata/src/main/java/com/redhat/lightblue/metadata/parser/MetadataParser.java

```diff
@@ -306,15 +307,17 @@ public Index parseIndex(T object) {
                 }
 
                 if (null != fields && !fields.isEmpty()) {
-                    List<SortKey> f = new ArrayList<>();
+                    List<IndexSortKey> f = new ArrayList<>();
 
                     for (T s : fields) {
                         String fld = getRequiredStringProperty(s, "field");
                         String dir = getStringProperty(s, "dir");
+                        String ci = getStringProperty(s, "caseInsensitive");
```

Comment:
Are you using "true"/"false" in your tests? Change that to true/false, and it won't throw typecast exception.